src/bootstrap/bootstrap.py

```diff
@@ -760,7 +763,13 @@ def main():
     help_triggered = (
         '-h' in sys.argv) or ('--help' in sys.argv) or (len(sys.argv) == 1)
     try:
-        bootstrap()
+        # If the user is asking for help, let them know that the whole download-and-build
+        # process has to happen before anything is printed out.
+        if help_triggered:
```

Comment:
Could you move this into `bootstrap()`? Top level `main` is for catching and handling exceptions only.